icc-x-api/icc-message-x-api.ts

```diff
@@ -290,17 +290,28 @@ export class IccMessageXApi extends iccMessageApi {
                       }))
                     )
                   }
-                  if (latestAction && !latestAction.closure) {
-                    const rp =
-                      phcp.referralPeriods && phcp.referralPeriods.find(per => !per.endDate)
-                    rp &&
-                      (rp.endDate = latestAction.date)(
-                        phcp.referralPeriods || (phcp.referralPeriods = [])
-                      ).push(new ReferralPeriod({ startDate: latestAction.date }))
-                  } else if (latestAction && latestAction.closure) {
-                    const rp =
-                      phcp && phcp.referralPeriods && phcp.referralPeriods.find(per => !per.endDate)
-                    rp && (rp.endDate = latestAction.date)
+                  if (!phcp.referralPeriods) {
+                    phcp.referralPeriods = []
+                  }
+
+                  const rp =
+                    (phcp.referralPeriods && phcp.referralPeriods.find(per => !per.endDate)) ||
+                    (phcp.referralPeriods[phcp.referralPeriods.length] = new ReferralPeriod({}))
+
+                  const actionDate = Number(
+                    moment(latestAction.date, "DD/MM/YYYY").format("YYYYMMDD")
```

Comment:
@phil1618 It is my understanding that we change the format of the date from `DD/MM/YYYY` to `YYYYMMDD `